packages/searchbase/src/utils.js

```diff
@@ -494,3 +494,89 @@ export function btoa(input = '') {
 
   return output;
 }
+
+export const componentsAlias = {
+  SEARCHBASE: 'SearchBase',
+  SEARCHBOX: 'SearchBox',
+  SEARCHCOMPONENT: 'SearchComponent'
+};
+
+export const backendAlias = {
+  MONGODB: 'mongodb', // mongodb
+  ELASTICSEARCH: 'elasticsearch' // elasticsearch
+};
+
+export const dataTypes = {
+  ARRAY: 'array',
+  FUNCTION: 'function',
+  OBJECT: 'object',
+  NUMBER: 'number',
+  BOOLEAN: 'boolean',
+  STRING: 'string'
+};
+
+const checkDataType = temp => {
+  if (typeof temp === dataTypes.OBJECT) {
+    if (Array.isArray(temp)) {
+      return dataTypes.ARRAY;
+    }
+
+    return dataTypes.OBJECT;
+  }
+  return typeof temp;
+};
+
+export function validateSchema(
+  passedProperties = {},
+  schema = {},
+  backendName = '',
+  componentName = ''
+) {
+  const passedPropertiesKeys = Object.keys(passedProperties).filter(
+    propertyKey => !!passedProperties[propertyKey]
+  );
+  const schemaPropertiesKeys = Object.keys(schema);
+  const requiredProperties = [];
+  const acceptedProperties = [];
+  // fetch required properties
+  schemaPropertiesKeys.forEach(propName => {
+    const currentProperty = schema[propName];
+    if (currentProperty.components.includes(componentName)) {
```

Comment:
Use `SearchBox` to just display the error message and remove it from the components list of schema.